### PR TITLE
dependencies: add libyaml

### PR DESCRIPTION
**Pull request by juliaazziz**
Add libyaml as a dependency. I found this necessary when compiling a specific version of OpenOCD.

### CODE REVIEW

File: dependencies/libyaml.sh [hunk 1]
```diff
@@ -0,0 +1,116 @@
+function libyaml_build()
+{
+  echo_develop
+  echo_develop "[${FUNCNAME[0]} $@]"
+
+  local libyaml_version="0.2.5"
```
Comment (ilg-ul) [inline dependencies/libyaml.sh]: Please use "$1" to pass the version from the caller.

File: dependencies/libyaml.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  local libyaml_src_folder_name="yaml-${libyaml_version}"
+
+  local libyaml_archive="${libyaml_src_folder_name}.tar.gz"
+
+  libyaml_url="https://pyyaml.org/download/libyaml/yaml-0.2.5.tar.gz"
```
Comment (ilg-ul) [inline dependencies/libyaml.sh]: Please use the variable.

**Comment by ilg-ul** [comment]
Does the build pass on all supported platforms?

**Comment by juliaazziz** [comment]
I just fixed the issue with the libyaml version that you pointed out! The build passes correctly.

**Comment by ilg-ul** [comment]
Thank you!

**Pull request merged**